src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -344,24 +352,32 @@ def __new_assessment_summary_json(self, assessment_packages_json, status, code):
         # discern started by - either pure auto-assessment or assessment data being included with configure patching with assessmentMode set to AutomaticByPlatform
         started_by = Constants.PatchAssessmentSummaryStartedBy.PLATFORM if (self.execution_config.exec_auto_assess_only or self.execution_config.include_assessment_with_configure_patching) else Constants.PatchAssessmentSummaryStartedBy.USER
 
-        # Compose sub-status message
-        substatus_message = {
-            "assessmentActivityId": str(self.execution_config.activity_id),
-            "rebootPending": self.is_reboot_pending,
-            "criticalAndSecurityPatchCount": critsec_patch_count,
-            "otherPatchCount": other_patch_count,
-            "patches": assessment_packages_json,
-            "startTime": str(self.execution_config.start_time),
-            "lastModifiedTime": str(self.env_layer.datetime.timestamp()),
-            "startedBy": str(started_by),
-            "errors": self.__set_errors_json(self.__assessment_total_error_count, self.__assessment_errors)
-        }
+        # Compose substatus message
+        errors = self.__set_errors_json(self.__assessment_total_error_count, self.__assessment_errors)
+
+        substatus_message = self.__compose_assessment_substatus_msg(
+            activity_id=self.execution_config.activity_id, reboot_pending=self.is_reboot_pending, crit_patch_count=critsec_patch_count,
+            other_patch_count=other_patch_count, packages=assessment_packages_json, start_time=self.execution_config.start_time,
+            last_modified_time=self.env_layer.datetime.timestamp(), started_by=started_by, errors=errors)
 
         if self.vm_cloud_type == Constants.VMCloudType.ARC:
             substatus_message["patchAssessmentStatus"] = code
             substatus_message["patchAssessmentStatusString"] = status
         return substatus_message
 
+    def __compose_assessment_substatus_msg(self, activity_id, reboot_pending, crit_patch_count, other_patch_count, packages, start_time, last_modified_time, started_by, errors):
+        return {
+            "assessmentActivityId": str(activity_id),
+            "rebootPending": reboot_pending,
+            "criticalAndSecurityPatchCount": crit_patch_count,
+            "otherPatchCount": other_patch_count,
+            "patches": packages,
+            "startTime": str(start_time),
+            "lastModifiedTime": str(last_modified_time),
+            "startedBy": str(started_by),
+            "errors": errors
+        }
+
```

Comment:
this refactoring was needed at the time when I was recomposing the assessment and installation substatus, then I modified the recompose to be shorter, so the refactor method wasn't apply anymore. I left the changes as it is because i see the benefit of having it extracted out in a method that it can be use in truncation logic again or else where

Reply:
Please refactor it in the PR that first uses this structure of calling - thereby justifying the change in the context of actual use. If no future PR is going to rely on this structure, then the programming principle **YAGNI** applies - https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it

When a significantly complex group of changes are going in, it's also important to reduce the surface area for investigation and debugging later if something goes wrong, and also to reduce load on code reviewers (which has been emphasized a few times already). 

Please revert this and any large change that's unnecessary to complete the functional aspects of what is trying to be achieved. If a subsequent PR justifies the change, we can review it in that PR.

Reply:
done